snappy-core/src/main/scala/org/apache/spark/sql/streaming/StreamBaseRelation.scala

```diff
@@ -79,6 +79,9 @@ abstract class StreamBaseRelation(options: Map[String, String])
   override val needConversion: Boolean = false
 
   override def buildScan(): RDD[Row] = {
+    if (context.getState() == StreamingContextState.STOPPED ){
```

Comment:
We need to discuss this more. If StreamingContext is stopped, how we are going to stop , clean-up other artefacts. 
One is catalog cleanup. What about already running queries. We should see how Spark jobs are handled on a stopped streaming context ( without stopping SparkContext)